test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -855,6 +839,214 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
+    m_all_eps.resize(2 * m_clients_num);
+
+    /* save all the clients' and server's eps in the m_all_eps array */
+    for (i = 0; i < m_clients_num; ++i) {
+        /* first 2 entities are m_server and m_client */
+        m_all_eps[i].ep                    = m_entities.at(2 + i).ep(0);
+        m_all_eps[i].state                 = 0;
+        m_all_eps[m_clients_num + i].ep    = m_server->ep(i);
+        m_all_eps[m_clients_num + i].state = 0;
+    }
+
+    /* Disconnect */
+    /* go over the eps array and for each ep - use rand() to decide whether or
+     * not it should initiate a disconnect */
+    for (i = 0; i < (2 * m_clients_num); ++i) {
+        if (ucs::rand() % 2) {
+            /* don't start a disconnect on an ep that was already disconnected */
+            pthread_mutex_lock(&m_lock);
+            if (!(m_all_eps[i].state & TEST_EP_FLAG_DISCONNECT_CB_INVOKED)) {
+                m_all_eps[i].state |= TEST_EP_FLAG_DISCONNECT_INITIATOR;
+                ASSERT_UCS_OK(uct_ep_disconnect(m_all_eps[i].ep, 0));
+                /* count the number of eps that initiated a disconnect */
+                m_ep_init_disconnect_cnt++;
+            }
+            pthread_mutex_unlock(&m_lock);
+        }
+    }
+
+    /* wait for all the disconnect flows that began, to complete.
+     * if an ep initiated a disconnect, its disconnect callback should have been
+     * called, and so is the disconnect callback of its remote peer ep.
+     * every ep that initiated a disconnect is counted. this couneter is
+     * decremented in its disconnect cb, therefore once all eps that initiated
+     * a disconnect are disconnected, this counter should be equal to zero */
+    deadline = ucs_get_time() + ucs_time_from_sec(10 * DEFAULT_TIMEOUT_SEC) *
+                                ucs::test_time_multiplier();
+
+    while ((m_ep_init_disconnect_cnt != 0) && (ucs_get_time() < deadline)) {
+        progress();
+    }
+    EXPECT_EQ(0, m_ep_init_disconnect_cnt);
+
+    /* count and print the number of eps that were not disconnected */
+    for (i = 0; i < (2 * m_clients_num); i++) {
+        if (m_all_eps[i].state == 0) {
+            no_disconnect_eps_cnt++;
+        } else {
+            EXPECT_TRUE((m_all_eps[i].state ==
+                        (TEST_EP_FLAG_DISCONNECT_INITIATOR |
+                         TEST_EP_FLAG_DISCONNECT_CB_INVOKED)) ||
+                        (m_all_eps[i].state == TEST_EP_FLAG_DISCONNECT_CB_INVOKED));
+        }
+    }
+
+    UCS_TEST_MESSAGE << no_disconnect_eps_cnt <<
+                     " (out of " << (2 * m_clients_num) << ") "
+                     "eps were not disconnected during the test.";
+
+    wait_for_client_server_counters(&m_server_disconnect_cnt,
+                                    &m_client_disconnect_cnt,
+                                    ((2 * m_clients_num) - no_disconnect_eps_cnt) / 2);
+
+    EXPECT_EQ(((2 * m_clients_num) - no_disconnect_eps_cnt) / 2, m_server_disconnect_cnt);
```

Comment:
what does the "((2 * m_clients_num) - no_disconnect_eps_cnt) / 2" mean?
can we save it into local variable since it's used 3 times?